flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/LaunchableMesosWorker.java

```diff
@@ -332,6 +334,22 @@ public String toString() {
 		return taskInfo.build();
 	}
 
+	/**
+	 * Get port keys representing the TM's configured endpoints. This includes mandatory TM endpoints such as
+	 * data and rpc as well as optionally configured endpoints for services such as prometheus reporter
+	 *
+	 * @return A deterministicly ordered Set of port keys to expose from the TM container
+	 */
+	private Set<String> getPortKeys() {
+		LinkedHashSet<String> tmPortKeys = new LinkedHashSet<>(Arrays.asList(TM_PORT_KEYS));
+		containerSpec.getDynamicConfiguration().keySet().stream()
+			.filter(key -> key.endsWith(".port") || key.endsWith(".ports"))  // This matches property naming convention
```

Comment:
You have to check the configured value as well, I know of at least one case where negative ports are used to disable features (and we wouldn't want enable these again).
Are there any use-cases where one might _not_ want mesos to provide the port?

Reply:
I agree that simply taking all configuration values which end with `port` and `ports` is problematic. What about configuration values whose value is needed, e.g. remote ports, and should not be overwritten. For example the `jobmanager.rpc.port` is one of these configuration values.

I would rather prefer a mesos specific configuration value which we can use to define whose ports need to be dynamically assigned. For example `mesos.dynamic-port-assignment: "metrics.prom.port, taskmanager.rpc.port, taskmanager.data.port"`. What do you think?